core/types/src/macros.rs

```diff
@@ -42,6 +42,23 @@ macro_rules! new_type_accessors_impls {
     )*}
 }
 
+/// Newtype wrapper for a primitive or struct type
+#[macro_export]
+macro_rules! impl_newtype {
```

Comment:
why is this not `new_type` ? 🤔

Reply:
I'll update the description of the PR,
It looks like all the rust docs treat it as one word [newtype](https://doc.rust-lang.org/book/ch19-04-advanced-types.html?#using-the-newtype-pattern-for-type-safety-and-abstraction)